### PR TITLE
Use instance for GPS logging / fix blended logging

### PR DESCRIPTION
**Pull request by peterbarker**
The blended logging is based on @amilcarlucas 's work here: https://github.com/ArduPilot/ardupilot/pull/15152/files

I've dropped the patch which validated the log IDs as no longer required from that PR.

Also corrects when we log this; it was logging way too often in that PR.

Output in SITL:
```
2020-11-19 12:59:13.17: GPS {TimeUS : 366213456, I : 2, Status : 6, GMS : 352771200, GWk : 2132, NSats : 10, HDop : 1.21, Lat : -35.3632621, Lng : 149.1652374, Alt : 584.09, Spd : 0.0, GCrs : 0.0, VZ : 0.0, Yaw : 0.0, U : 1}
2020-11-19 12:59:13.37: GPS {TimeUS : 366413376, I : 0, Status : 6, GMS : 352771400, GWk : 2132, NSats : 10, HDop : 1.21, Lat : -35.3632621, Lng : 149.1652374, Alt : 584.09, Spd : 0.0, GCrs : 0.0, VZ : 0.0, Yaw : 0.0, U : 0}
2020-11-19 12:59:13.37: GPS {TimeUS : 366413376, I : 1, Status : 6, GMS : 352771400, GWk : 2132, NSats : 10, HDop : 1.21, Lat : -35.3632621, Lng : 149.1652374, Alt : 584.09, Spd : 0.0, GCrs : 0.0, VZ : 0.0, Yaw : 0.0, U : 0}
2020-11-19 12:59:13.37: GPS {TimeUS : 366413376, I : 2, Status : 6, GMS : 352771400, GWk : 2132, NSats : 10, HDop : 1.21, Lat : -35.3632621, Lng : 149.1652374, Alt : 584.09, Spd : 0.0, GCrs : 0.0, VZ : 0.0, Yaw : 0.0, U : 1}
2020-11-19 12:59:13.57: GPS {TimeUS : 366613296, I : 0, Status : 6, GMS : 352771600, GWk : 2132, NSats : 10, HDop : 1.21, Lat : -35.3632621, Lng : 149.1652374, Alt : 584.09, Spd : 0.0, GCrs : 0.0, VZ : 0.0, Yaw : 0.0, U : 0}
2020-11-19 12:59:13.57: GPS {TimeUS : 366613296, I : 1, Status : 6, GMS : 352771600, GWk : 2132, NSats : 10, HDop : 1.21, Lat : -35.3632621, Lng : 149.1652374, Alt : 584.09, Spd : 0.0, GCrs : 0.0, VZ : 0.0, Yaw : 0.0, U : 0}
2020-11-19 12:59:13.57: GPS {TimeUS : 366613296, I : 2, Status : 6, GMS : 352771600, GWk : 2132, NSats : 10, HDop : 1.21, Lat : -35.3632621, Lng : 149.1652374, Alt : 584.09, Spd : 0.0, GCrs : 0.0, VZ : 0.0, Yaw : 0.0, U : 1}
```

### CODE REVIEW

**Review by amilcarlucas — APPROVED** [review]
Nice cleanup!!!

File: libraries/AP_GPS/AP_GPS.cpp [hunk 1]
```diff
@@ -1659,6 +1659,10 @@ void AP_GPS::calc_blended_state(void)
     _blended_antenna_offset.zero();
     _blended_lag_sec = 0;
 
+#ifndef HAL_BUILD_AP_PERIPH
+    const uint32_t last_blended_message_time_ms = timing[GPS_BLENDED_INSTANCE].last_message_time_ms;
+    bool data_should_be_logged = false;
```
Comment (WickedShell) [inline libraries/AP_GPS/AP_GPS.cpp:1664]: We don't need the bool.

File: libraries/AP_GPS/AP_GPS.cpp [hunk 2]
```diff
@@ -1801,6 +1804,16 @@ void AP_GPS::calc_blended_state(void)
     }
     timing[GPS_BLENDED_INSTANCE].last_fix_time_ms = (uint32_t)temp_time_1;
     timing[GPS_BLENDED_INSTANCE].last_message_time_ms = (uint32_t)temp_time_2;
+
+#ifndef HAL_BUILD_AP_PERIPH
+    if (timing[GPS_BLENDED_INSTANCE].last_message_time_ms > last_blended_message_time_ms) {
```
Comment (WickedShell) [inline libraries/AP_GPS/AP_GPS.cpp:1809]: Just use this as the first line of the if on line 1812, and remove the `data_should_be_logged` bool. It's just increasing the complexity here (and maybe the code size, but hopefully the compiler is managing this)

**Review by WickedShell — CHANGES_REQUESTED** [review]

File: libraries/AP_GPS/AP_GPS.cpp (hunk 1 above)
Comment (peterbarker) [reply libraries/AP_GPS/AP_GPS.cpp]: Fixed.

File: libraries/AP_GPS/AP_GPS.cpp (hunk 2 above)
Comment (peterbarker) [reply libraries/AP_GPS/AP_GPS.cpp]: Yep.

**Review by WickedShell — APPROVED** [review]

**Pull request merged**